pyproject.toml

```diff
@@ -16,7 +16,7 @@ packages = [
 "Bug Tracker" = "https://github.com/temporalio/samples-python/issues"
 
 [tool.poetry.dependencies]
-python = "^3.7"
+python = ">=3.8.1,<4.0"
```

Comment:
I think we should have a PR first that just drops 3.7 support and moves CI from 3.7 + 3.11 to 3.8 + 3.12. We did this for the primary SDK at https://github.com/temporalio/sdk-python/pull/422. If you want to make that part of this issue, I'd be ok with that too (but change the title too).

Reply:
Already done as part of #101. This PR needs to be updated.